app/battle.py

```diff
@@ -0,0 +1,31 @@
+from typing import List
+from app.knights import Knight
+
+
+class BattleKnight:
+    def __init__(self, knights: List[Knight]):
+        self.knights = knights
+
+    def battle_preparation(self) -> None:
+        for knight in self.knights:
+            knight.protection = sum(armour["protection"] for armour in knight.armour)
```

Comment:
Ensure that the `armour` attribute in the `Knight` class is a list of dictionaries with a `protection` key, as assumed here. This is consistent with the provided configuration, but double-check the `Knight` class definition to confirm.